src/vm/siginfo.cpp

```diff
@@ -4987,16 +5030,21 @@ void ReportPointersFromValueType(promote_func *fn, ScanContext *sc, PTR_MethodTa
 void ReportPointersFromValueTypeArg(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, ArgDestination *pSrc)
 {
     WRAPPER_NO_CONTRACT;
-    
-    if (!pMT->ContainsPointers())
-        return;
+
+    if(pMT->ContainsPointers())
```

Comment:
As I currently do not have a complete understanding of GC holes with respect to Unix and structs passed in registers, I'm conservatively preserving the previous behavior by calling `ReportPointersFromStructInRegisters` only under the previous conditions. #9033 is intended to track potential issues on Unix, a fix for which should hopefully resolve the question.